PyInstaller/utils/hooks/__init__.py

```diff
@@ -1015,26 +1017,69 @@ def relpath_to_config_or_make(filename):
 
 
 def get_typelibs(module, version):
+    '''deprecated; only here for backwards compat'''
+    return get_gi_typelibs(module, version)[1]
+
+def get_gi_typelibs(module, version):
+    """
+    Returns a tuple of (binaries, datas, hiddenimports) to be used by PyGObject
+    related hooks. Loads dependencies recursively also.
+    """
     datas = []
+    binaries = []
+    hiddenimports = []
+    
+    _get_gi_typelibs(module, version, datas, binaries, hiddenimports)
+    
+    return binaries, datas, hiddenimports
+    
+def _get_gi_typelibs(module, version, datas, binaries, hiddenimports):
+    
     statement = """
 import gi
 gi.require_version("GIRepository", "2.0")
 from gi.repository import GIRepository
-print(GIRepository.Repository.get_search_path())"""
-    typelibs_path = eval_statement(statement)
-    if not typelibs_path:
+repo = GIRepository.Repository.get_default()
+module, version = ('%s', '%s')
```

Comment:
Should be `%r` (without quotes), do ensure correct quoting. Otherwise a string like `funny'string` will break (much like SQL-injection :-)